db/migrate/20200331183640_add_repository_snapshots.rb

```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+
+class AddRepositorySnapshots < ActiveRecord::Migration[6.0]
+  def up
+    add_column :repositories, :parent_id, :bigint, null: true
+    add_reference :repositories, :my_module
+    add_column :repositories, :type, :string
```

Comment:
Is 'type' really the best name for this? Won't it interfere with some reserved keywords?

Reply:
It is actually default column name for STI